engine/tester/config_parse.c

```diff
@@ -996,27 +997,122 @@ get_test_attrs(xmlNodePtr node, test_attrs *attrs)
     return 0;
 }
 
+/**
+ * Get run template by name.
+ *
+ * @param name      Template name
+ * @param session   Session the run item belongs to
+ * @param run_templ Location of run template item
+ *
+ * @return Status code.
+ */
+static te_errno
+get_template_by_name(const char *name, const test_session *session,
+                     const run_item **run_tmpl)
+{
+    run_item *rtmpl;
+    const test_session *s = session;
+
+    while (s != NULL)
+    {
+        TAILQ_FOREACH(rtmpl, &s->templates, links)
+        {
+            if (strcmp(rtmpl->name, name) == 0)
+            {
+                *run_tmpl = rtmpl;
+                return 0;
+            }
+        }
+        s = s->parent;
+    }
+    ERROR("There is no run template with name '%s'", name);
+    return TE_RC(TE_TESTER, TE_ENOENT);
+}
+
+/**
+ * Copy arguments from run template.
+ *
+ * @param run_templ Location of run template item
+ * @param list      List to copy arguments to it
+ *
+ * @return Status code.
+ */
+static void
+copy_template_args(const run_item *tmpl, test_vars_args *list)
+{
+    test_var_arg    *rtarg;
+    test_var_arg    *p;
+
+    if (tmpl == NULL)
+        return;
+
+    TAILQ_FOREACH(rtarg, &tmpl->args, links)
+    {
+        p = TE_MEMDUP(rtarg, sizeof(*rtarg));
+        p->tmpl_arg = true;
+        TAILQ_INSERT_TAIL(list, p, links);
+    }
+
+    return;
+}
+
+/**
+ * Add argument to the appropriate place in the list.
+ *
+ * @param p         Argument
+ * @param list      List to add argument to it
+ *
+ * @return Status code.
+ */
+void
+add_or_fix_arg(test_var_arg *p, test_vars_args *list)
+{
+    test_var_arg *arg;
+    TAILQ_FOREACH(arg, list, links)
+    {
+        if (strcmp(arg->name, p->name) == 0)
+        {
+            TAILQ_INSERT_AFTER(list, arg, p, links);
+            TAILQ_REMOVE(list, arg, links);
+            return;
+        }
+    }
+    TAILQ_INSERT_TAIL(list, p, links);
+    return;
+}
 
 /**
  * Get script call description.
  *
- * @param node      XML node with script call description
- * @param cfg       Tester configuration context
- * @param script    Location for script call description
+ * @param node         XML node with script call description
+ * @param cfg          Tester configuration context
+ * @param ritem        Run item with the script
  *
  * @return Status code.
  */
 static te_errno
-get_script(xmlNodePtr node, tester_cfg *cfg, test_script *script)
+get_script(xmlNodePtr node, tester_cfg *cfg, run_item *ritem)
 {
     te_errno            rc;
     const test_info    *ti;
     bool                objective_found = false;
     bool                execute_found = false;
+    test_script        *script = &ritem->u.script;
+    const test_script  *tmpl_script = NULL;
+
+#define ADD_FROM_TMPL(_field) \
+    do {                                                              \
+        if (script->_field == NULL && ritem->tmpl != NULL)            \
```

Comment:
it is better to check tmpl_script instead of ritem->tmpl since tmpl_script should be used below.